src/jit/hwintrinsicxarch.cpp

```diff
@@ -237,10 +236,7 @@ int HWIntrinsicInfo::lookupNumArgs(const GenTreeHWIntrinsic* node)
             list = list->Rest();
         } while (list != nullptr);
 
-        assert(numArgs == 3);
-#endif
-
-        return 3;
+        return numArgs;
```

Comment:
`GatherMaskVector*` have 5 parameters that breaks our previous `numArgs` assumption.

Reply:
Do we actually have to carry all 5 args through? They eventually get folded down to a `target register`, `vm32x`, and `mask register`, so I would think we we could do some of this folding in the importer.

Reply:
> so I would think we we could do some of this folding in the importer.

That seems to require a new IR to contain the folded VSIB.

Reply:
Yeah, I guess we could only fold it to 3 if we knew the register for `index` before-hand....

It just seems bad to make this many changes for a single family of instructions, since it will impact the 90% case (which is the other HWIntrinsics)...